projects/igniteui-angular/src/lib/data-operations/filtering-util.ts

```diff
@@ -0,0 +1,127 @@
+import { IgxBooleanFilteringOperand, IgxDateFilteringOperand, IgxDateTimeFilteringOperand, IgxFilteringOperand, IgxNumberFilteringOperand, IgxStringFilteringOperand, IgxTimeFilteringOperand } from './filtering-condition';
+import { IFilteringExpression, ISerializedFilteringExpression } from './filtering-expression.interface';
+import { FilteringExpressionsTree, IFilteringExpressionsTree, ISerializedFilteringExpressionTree } from './filtering-expressions-tree';
+
+export class FilteringUtil {
+    /**
+     * Recreates a `FilteringExpressionsTree` from a given JSON serialized string.
+     * @param jsonString The serialized string to recreate the tree from.
+     * @returns A fully reconstructed `FilteringExpressionsTree`.
+     */
+    public static recreateTreeFromJson(jsonString: string): FilteringExpressionsTree {
+        return this.recreateTree(JSON.parse(jsonString));
+    }
+
+    /**
+     * Recreates a `FilteringExpressionsTree` from its serialized model.
+     * @param model The serialized model to recreate the tree from.
+     * @returns A fully reconstructed `FilteringExpressionsTree`.
+     */
+    public static recreateTree(model: ISerializedFilteringExpressionTree): FilteringExpressionsTree {
```

Comment:
This method does not return properly a condition for a Boolean operand in an inner query.

In the dev demo add the following code and compare the two trees:
```
var treeS = FilteringUtil.recreateTree((this.expressionTree as FilteringExpressionsTree).toJSON());
var treeSJ = FilteringUtil.recreateTreeFromJson(JSON.stringify((this.expressionTree as FilteringExpressionsTree).toJSON()));
```
![image](https://github.com/user-attachments/assets/3477f7e7-189b-4bd0-8a9a-41c6fa9d3d8f)

Reply:
I've resolved this in the latest commit.